File: BinaryTreeRightSideView/BinaryTreeRightSideView.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        vector<int> result;
        if(root == nullptr)
        	return result;
        TreeNode* pre = root;
        queue<TreeNode*> tmp;
        tmp.push(pre);
        while(!tmp.empty()) {
        	int cnt = tmp.size();
        	for(int i = 0; i < cnt; i++) {
        		TreeNode* cur = tmp.front();
        		tmp.pop();
        		if(cur->left)
        			tmp.push(cur->left);
        		if(cur->right)
        			tmp.push(cur->right);
        		if(i == cnt - 1)
        			result.push_back(cur->val);
        	}
        }
        return result;
    }
};

File: BinaryTreeRightSideView/BinaryTreeRightSideView.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def rightSideView(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        res = []
        if root is None:
            return res
        pre = root
        tmp = []
        tmp.append(pre)
        while len(tmp) > 0:
            cnt = len(tmp)
            for i in range(cnt):
                cur = tmp[0]
                tmp.remove(cur)
                if cur.left:
                    tmp.append(cur.left)
                if cur.right:
                    tmp.append(cur.right)
                
            res.append(cur.val)
        return res